src/CVRF/cvrf_priv.c

```diff
@@ -0,0 +1,2292 @@
+	ret->doc_references = oscap_list_new();
+	ret->acknowledgments = oscap_list_new();
+	return ret;
+}
+
+void cvrf_document_free(struct cvrf_document *doc) {
+	if (doc == NULL)
+		return;
+
+	oscap_free(doc->doc_distribution);
+	oscap_free(doc->aggregate_severity);
+	oscap_free(doc->namespace);
+	cvrf_doc_tracking_free(doc->tracking);
+	cvrf_doc_publisher_free(doc->publisher);
+	oscap_list_free(doc->doc_references, (oscap_destruct_func) cvrf_reference_free);
+	oscap_list_free(doc->acknowledgments, (oscap_destruct_func) cvrf_acknowledgment_free);
+	oscap_free(doc);
+}
+
+struct cvrf_document *cvrf_document_clone(const struct cvrf_document *doc) {
+	struct cvrf_document *clone = oscap_calloc(1, sizeof(struct cvrf_document));
+	clone->doc_distribution = oscap_strdup(doc->doc_distribution);
+	clone->aggregate_severity = oscap_strdup(doc->aggregate_severity);
+	clone->namespace = oscap_strdup(doc->namespace);
+	clone->tracking = cvrf_doc_tracking_clone(doc->tracking);
+	clone->publisher = cvrf_doc_publisher_clone(doc->publisher);
+	clone->doc_references = oscap_list_clone(doc->doc_references, (oscap_clone_func) cvrf_reference_clone);
+	clone->acknowledgments = oscap_list_clone(doc->acknowledgments, (oscap_clone_func) cvrf_acknowledgment_clone);
+	return clone;
+}
+
+/***************************************************************************
+ * CVRF Model
+ * Top-level structure of the CVRF hierarchy
+ */
+struct cvrf_model {
+	char *doc_title;
+	char *doc_type;
+	struct cvrf_document *document;
+	struct cvrf_product_tree *tree;
+	struct oscap_list *vulnerabilities;	/* 1-n */
+};
+OSCAP_ACCESSOR_STRING(cvrf_model, doc_title)
+OSCAP_ACCESSOR_STRING(cvrf_model, doc_type)
+OSCAP_IGETINS_GEN(cvrf_vulnerability, cvrf_model, vulnerabilities, vulnerability)
+OSCAP_ITERATOR_REMOVE_F(cvrf_vulnerability)
+
+struct cvrf_product_tree *cvrf_model_get_product_tree(struct cvrf_model *model) {
+	return model->tree;
+}
+struct cvrf_document *cvrf_model_get_document(struct cvrf_model *model) {
+	return model->document;
+}
+const char *cvrf_model_get_identification(struct cvrf_model *model) {
+	struct cvrf_doc_tracking *tracking = cvrf_document_get_tracking(model->document);
+	return (cvrf_doc_tracking_get_tracking_id(tracking));
+}
+
+struct cvrf_model *cvrf_model_new() {
+	struct cvrf_model *ret = oscap_alloc(sizeof(struct cvrf_model));
+	if (ret == NULL)
+		return NULL;
+
+	ret->doc_title = NULL;
+	ret->doc_type = NULL;
+	ret->document = cvrf_document_new();
+	ret->tree = cvrf_product_tree_new();
+	ret->vulnerabilities = oscap_list_new();
+	return ret;
+}
+
+void cvrf_model_free(struct cvrf_model *cvrf) {
+	if (cvrf == NULL)
+		return;
+
+	oscap_free(cvrf->doc_title);
+	oscap_free(cvrf->doc_type);
+	cvrf_document_free(cvrf->document);
+	cvrf_product_tree_free(cvrf->tree);
+	oscap_list_free(cvrf->vulnerabilities, (oscap_destruct_func) cvrf_vulnerability_free);
+	oscap_free(cvrf);
+}
+
+void cvrf_model_clone(struct cvrf_model *clone, const struct cvrf_model *model) {
```

Comment:
Shouldn't this create the clone? What is the user of the API expected to pass in?